c10/core/StorageImpl.h

```diff
@@ -5,8 +5,12 @@
 
 #include <c10/util/intrusive_ptr.h>
 
+#include <atomic>
+
 namespace c10 {
 
+static std::atomic<std::uint64_t> next_unique_storage_id{0};
```

Comment:
I am wary of using a global to track this information and add an additional 8 bytes to `StorageImpl`. It also seems to be a local answer to a more fundamental question, which is: how do we handle identity for torch objects? 

So far, we have mostly followed [Python's convention](https://docs.python.org/3/library/functions.html#id) and used the pointer. This is also how we do `id()`/`hash()` for Python tensors and their storages and thus for their serialization.

This obviously has serious issues if we are re-using memory, but has the advantage that it's consistent across all objects and matches what Python does. @ezyang any thoughts here?